internal/kibana/role.go

```diff
@@ -370,6 +436,66 @@ func expandKibanaRoleElasticsearch(v interface{}) *kbapi.KibanaRoleElasticsearch
 				elasticConfig.Indices = indices
 			}
 
+			if v, ok := userElasticConfig["remote_indices"]; ok {
```

Comment:
We'll want to throw an error if this attribute is configured, but not supported by the underlying cluster. The [index](https://github.com/elastic/terraform-provider-elasticstack/blob/1c23072930f15b03c8548a4ba215e6d16bf98b24/internal/elasticsearch/index/index.go#L776-L781) resource has an example of doing that. 

We'll also need to update the acceptance tests to only configure `remote_indices` when supported.